### PR TITLE
texmaker: use wrapQtAppsHook

### PR DESCRIPTION
**Pull request by cruegge**
###### Motivation for this change

Texmaker on NixOS 19.09 is currently broken due to missing `wrapQtAppsHook`. I successfully tested the patch on the `nixpkgs-channels/nixos-19.09` branch (at `80b42e6`), but it fails to compile on master, presumably for unrelated reasons. I still rebased it according to the guidelines.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @markuskowa @Chaddai

### CODE REVIEW

File: pkgs/applications/editors/texmaker/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [ qtbase qtscript poppler zlib ];
-  nativeBuildInputs = [ pkgconfig poppler qmake ];
+  nativeBuildInputs = [ pkgconfig poppler qmake wrapQtAppsHook ];
```
Comment (markuskowa) [inline pkgs/applications/editors/texmaker/default.nix]: I think you do not need both. Using Qt5's make derivation should be sufficient.

Comment (cruegge) [reply pkgs/applications/editors/texmaker/default.nix]: Ok, I'll update the PR.

File: pkgs/applications/editors/texmaker/default.nix [hunk 2]
```diff
@@ -1,6 +1,6 @@
-{ stdenv, fetchurl, qtbase, qtscript, qmake, zlib, pkgconfig, poppler }:
+{ stdenv, mkDerivation, fetchurl, qtbase, qtscript, qmake, zlib, pkgconfig, poppler }:
```
Comment (markuskowa) [inline pkgs/applications/editors/texmaker/default.nix]:
```suggestion
{ lib, mkDerivation, fetchurl, qtbase, qtscript, qmake, zlib, pkgconfig, poppler }:
```

Comment (markuskowa) [reply pkgs/applications/editors/texmaker/default.nix]: (and  please fix the `meta = with stdenv.lib`)

**Comment by markuskowa** [comment]
@GrahamcOfBorg build texmaker

**Comment by cruegge** [comment]
Build failures are probably related to #71463. Should I wait for that to get resolved and rebase again?

**Comment by markuskowa** [comment]
I am not sure this is related since this is the error: `<command-line>: error: missing terminating " character` but I can confirm that this PR fixes the build on 19.09.

**Comment by cruegge** [comment]
If I change the texmaker derivation along the lines of the patch mentioned int he other issue (83709b0), i.e. replace the `preConfigure` stuff by `qmakeFlags` and `placeholder`, it builds on master, and the result does contain the desktop file, icon and metainfo. I suppose there is some broken shell expansion / quoting to blame here.

I could add another commit that fixes this for texmaker (avoiding shell quoting is a good idea anyway), or open a new PR.

**Comment by markuskowa** [comment]
A second commit that fixes it would be great.

**Comment by markuskowa** [comment]
Thanks for the fix!

**Comment by markuskowa** [comment]
cherry picked for 19.09 in 56c4063ce6b7e2973f123f6552354a2a80dfc7d6

**Pull request merged**